### PR TITLE
rbd: don't attempt to interpret image cache state json

### PR DESCRIPTION
**Pull request by idryomov**
There are several issues with the current code:

- the value of "present" field is lost to the value of "clean" -- if the
  cache is present but dirty, we report that there is no cache
- the value of "clean" field is lost -- we always report false
- for a cache bigger than 2G, size is reported incorrectly -- 32-bit int
  overflow
- there is no way to get the type of the cache (rwl vs ssd)

Finally, even with all of the above fixed, image cache state output
would still be confusing because it is effectively a snapshot from
the time the cache was loaded.  It is not updated until the cache is
orderly closed.

Given that more changes would be needed to make "rbd status" output
useful and in the meantime it just needs to be correct to allow for
basic troubleshooting, just dump raw json.

Fixes: https://tracker.ceph.com/issues/50613
Signed-off-by: Ilya Dryomov <idryomov@gmail.com>

### CODE REVIEW

**Comment by idryomov** [comment]
cc @MahatiC

**Comment by idryomov** [comment]
cc @CongMinYin

**Comment by CongMinYin** [comment]
lgtm, thanks!

**Comment by CongMinYin** [comment]
lgtm, thanks!

File: src/tools/rbd/action/Status.cc [hunk 1]
```diff
@@ -186,13 +146,8 @@ static int do_show_status(librados::IoCtx& io_ctx, const std::string &image_name
...
     }
-    if (cache_state.present) {
-      f->open_object_section("image_cache_state");
-      f->dump_bool("clean", cache_state.clean);
-      f->dump_int("size", cache_state.size);
-      f->dump_string("host", cache_state.host);
-      f->dump_string("path", cache_state.path);
-      f->close_section(); // image_cache_state
+    if (!image_cache_str.empty()) {
+      f->dump_string("image_cache_state", image_cache_str);
```
Comment (MahatiC) [inline src/tools/rbd/action/Status.cc:150]: Can we add a `TODO` to fix this properly for future?

Comment (MahatiC) [reply src/tools/rbd/action/Status.cc:150]: https://tracker.ceph.com/issues/50614 - I saw this is already logged. Thanks.

**Comment by MahatiC** [comment]
thanks, lgtm!

**Pull request merged**